core/storage/email/gae_models_test.py

```diff
@@ -128,9 +149,131 @@ def test_get_by_hash_behavior_with_sent_datetime_lower_bound(self):
+                'user_id', 'thread_id')
+            created1.put()
+            created2 = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+                'user_id', 'thread_id')
+            created2.put()
+
+    def test_get_by_reply_to_id_works_correctly(self):
+        # Name is too long for linter.
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        created = model.create('user_id', 'thread_id')
+        created.put()
+
+        result = model.get_by_reply_to_id(created.reply_to_id)
+
+        self.assertNotEqual(result, None) #There should be a result
+        self.assertEqual(result.reply_to_id, created.reply_to_id)
+
+        result = model.get_by_reply_to_id('non_existent_reply_to_id')
+
+        self.assertEqual(result, None)
+
+    def test_get_works_correctly(self):
+        created = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'thread_id')
+        created.put()
+
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+            'user_id', 'thread_id', strict=False)
+
+        self.assertNotEqual(result, None)
+        self.assertIn('user_id.thread_id', str(result.key))
+
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+            'bad_user_id', 'bad_thread_id', strict=False) #Should not throw
+        self.assertEqual(result, None)
+
+        with self.assertRaises(Exception):
+            result = email_models.GeneralFeedbackEmailReplyToIdModel.get(
+                'bad_user_id', 'bad_thread_id') #Should throw
+
+    def test_get_multi_by_user_ids_works_correctly(self):
+        # Name is too long for linter.
+        model = email_models.GeneralFeedbackEmailReplyToIdModel
+
+        model.create('user_id_1', 'thread_id')
+        model.create('user_id_2', 'thread_id')
+
+        result = model.get_multi_by_user_ids(
+            ['user_id_1', 'user_id_2'], 'thread_id')
+
+        self.assertEqual(len(result), 2)
+        self.assertIn('user_id_1', result.keys())
+        self.assertIn('user_id_2', result.keys())
+
+    def test_user_id(self):
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'entity_type.entity_id.thread_id')
+
+        self.assertEqual(result.user_id, 'user_id')
+
+    def test_entity_type(self):
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'entity_type.entity_id.thread_id')
+
+        self.assertEqual(result.entity_type, 'entity_type')
+
+    def test_entity_id(self):
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'entity_type.entity_id.thread_id')
+
+        self.assertEqual(result.entity_id, 'entity_id')
+
+    def test_thread_id(self):
+        result = email_models.GeneralFeedbackEmailReplyToIdModel.create(
+            'user_id', 'entity_type.entity_id.thread_id')
+
+        self.assertEqual(result.thread_id, 'entity_type.entity_id.thread_id')
+
```

Comment:
The above four tests are failing. Here is the screen shot.

![screenshot from 2018-12-11 01-48-50](https://user-images.githubusercontent.com/9693472/49759032-8773f000-fce6-11e8-9f45-b85984f8b143.png)

thread_id, entity_id etc. are accessed but the class only has `reply_to_id` variable in the class. I don't know how these tests are passing on your system though.

Reply:
Well, I just removed the tests then. "Fixed"

Reply:
I think I know the reason here! #5940 recently moved these to the domain layer. So probably, your branch isn't up to date, and on updating the tests might fail locally for you too @CelineDknp. Then you can fix them! :)

Reply:
So yeah, these lines you are trying to cover no longer exist in emails/gae_models.py. So yeah, removing them is the right thing to do. Corresponding tests for the new functions were added in email_domain.py

Reply:
I fixed the test as suggested by @ankita240796, but I can remove them if it is better

Reply:
I would suggest that we remove them. The point of moving them to the domain layer was to prevent having complicated splits on the id in the model layer. Ideally, the model should just know that this is the id.

You have already tested specifics regarding the id like creation. So I think it is ok to remove these tests. The model layer need not know that user_id or entity_type, etc can be extracted. That will be handled further up the stack. 

Does this help?

Reply:
Understood ! Fixed, and the PR should be good now